### DISCUSSION TITLE
Edit Locally doesn't work

### DISCUSSION THREAD

**Discussion by gbertolottiKS** [nextcloud/all-in-one#1914] [Questions] [answered]
### Steps to reproduce
1. sync a folder with a local one via the Desktop app
2. select a file on Nextcloud and click _Edit locally_

### Expected behavior 
The file should be locked (to prevent modifications from others) and opened with the default application on the local machine. After the saving and the close of the file, it should sync the changes, and unlock the file on Nextcloud (as described here: https://nextcloud.com/blog/nextcloud-office-release-solves-document-compatibility-overhauls-knowledge-management/)

### Actual behavior 
It failed to lock the file, displaying this error:
![immagine](https://user-images.githubusercontent.com/115481064/216395125-a01baae2-eb74-4735-811a-cbeea15e6350.png)

### Host OS 
Ubuntu 22.04

#### Nextcloud AIO version 
4.2.0

#### Current channel 
latest

#### Other valuable info 
After some research, we found firstly a forum thread which suggest installing the Temporary Lock app (but, if it's mandatory, why is not installed by default?).
We made it, and after a nextcloud desktop app reboot, it seemed to work: the error message disappeared, but actually doesn't work.
When we select _Edit locally_, it locks the file and opens it locally, but after saving it, it fails to sync because it finds itself locked (_server reply "423 locked"_), and doesn't it unlock. 
![immagine](https://user-images.githubusercontent.com/115481064/216394204-79e8d1d9-5fe9-407a-93b4-342b2f061e34.png)
Even after manually unlock, nothing changes.

After more research, we found this issue: https://github.com/nextcloud/desktop/issues/5101, that seems pretty interesting; the only problem is that the change that was supposed to fix the problem seems to have been released with version 3.6.3, and now we have version 3.6.6, and it still doesn't work.

Any idea?
The Temporary Lock app is mandatory?
Thanks!

**Comment by szaimen** [comment, marked as answer]
> The Temporary Lock app is mandatory?

yes

**Reply by gbertolottiKS** [reply]
Ok. But does it need some configuration? The way it is now, it doesn't work.

**Reply by szaimen** [reply]
> But does it need some configuration?

Nothing I am aware of. Did you install the app?

Or what doesnt work?

**Reply by gbertolottiKS** [reply]
Yes, we install the app.
But, as I said above, the Edit Locally functionality is still unusable: it fails to sync the changes made locally (because of the "423 locked" error), and it does not unlock the file after its saving.
So, in the end, it does nothing more than lock us a file and create an error because synchronization has failed 😅. No changes made locally are saved

**Reply by szaimen** [reply]
Okay, then it would probably make sense to open an issue about this either in the files_lock or desktop repo.
